generate-environment-info/action.yml

```diff
@@ -16,70 +17,99 @@ inputs:
+      id: evaluate-environment
       shell: bash
       run: |
         echo "GITHUB_REF_TYPE = $GITHUB_REF_TYPE"
         echo "GITHUB_REF_NAME = $GITHUB_REF_NAME"
-        if [ $GITHUB_REF_TYPE == "branch" ]; then
-          if [ $GITHUB_REF_NAME == "dev" ]; then
-            echo "::set-output name=name::development"
-            echo "::set-output name=prefix::dev"
-            echo "::set-output name=version::dev-${{ steps.generate-sha.outputs.sha }}"
-            echo "::set-output name=rails-env::dev"
-            echo "::set-output name=eb-environment-names::${{ inputs.eb-environment-names-dev }}"
-          elif [ $GITHUB_REF_NAME == "qa" ]; then
-            echo "::set-output name=name::QA"
-            echo "::set-output name=prefix::qa"
-            echo "::set-output name=version::qa-${{ steps.generate-sha.outputs.sha }}"
-            echo "::set-output name=rails-env::qa"
-            echo "::set-output name=eb-environment-names::${{ inputs.eb-environment-names-qa }}"
-          elif [ $GITHUB_REF_NAME == "uat" ]; then
-            echo "::set-output name=name::UAT"
-            echo "::set-output name=prefix::uat"
-            echo "::set-output name=version::uat-${{ steps.generate-sha.outputs.sha }}"
-            echo "::set-output name=rails-env::uat"
-            echo "::set-output name=eb-environment-names::${{ inputs.eb-environment-names-uat }}"
-          elif [ $GITHUB_REF_NAME == "staging" ]; then
-            echo "::set-output name=name::staging"
-            echo "::set-output name=prefix::staging"
-            echo "::set-output name=version::staging-${{ steps.generate-sha.outputs.sha }}"
-            echo "::set-output name=rails-env::staging"
-            echo "::set-output name=eb-environment-names::${{ inputs.eb-environment-names-staging }}"
-          else
-            exit 1
-          fi
-        elif [ $GITHUB_REF_TYPE == "tag" ]; then
-          echo "::set-output name=name::production"
-          echo "::set-output name=prefix::prod"
-          echo "::set-output name=version::$GITHUB_REF_NAME"
-          echo "::set-output name=rails-env::production"
-          echo "::set-output name=eb-environment-names::${{ inputs.eb-environment-names-production }}"
+
+        branch_or_tag_regex="^(branch|tag)$"
+        # matches
+        semmantic_commit_regex="^v[0-9]+\.[0-9]+\.[0-9]+(-[a-z]+(\.[0-9]+)?)?$"
+
+        if [[ $GITHUB_REF_TYPE =~ $branch_or_tag_regex && $GITHUB_REF_NAME == "dev" ]]; then
+          environment=dev
+        elif [[ $GITHUB_REF_TYPE =~ $branch_or_tag_regex && $GITHUB_REF_NAME == "qa" ]]; then
+          environment=qa
+        elif [[ $GITHUB_REF_TYPE =~ $branch_or_tag_regex && $GITHUB_REF_NAME == "uat" ]]; then
+          environment=uat
+        elif [[ $GITHUB_REF_TYPE =~ $branch_or_tag_regex && $GITHUB_REF_NAME == "staging" ]]; then
+          environment=staging
+        elif [[ $GITHUB_REF_TYPE =~ $semmantic_commit_regex ]]; then # TODO: fix
+          environment=production
```

Comment:
I guess I deserved this, merging without waiting for second approval. Will fix in follow up PR.